lib/lightning/attempts.ex

```diff
@@ -232,4 +233,30 @@ defmodule Lightning.Attempts do
   defp adaptor do
     Lightning.Config.attempts_adaptor()
   end
+
+  def mark_unfinished_runs_lost(%Attempt{id: attempt_id}) do
+    # Fetch all runs associated with the attempt
+    runs =
+      Repo.all(
+        from ar in Lightning.AttemptRun,
+          where: ar.attempt_id == ^attempt_id,
+          preload: [:run]
+      )
+
+    # Iterate over each run and mark it as lost if not finished
+    Enum.each(runs, fn %{run: run} ->
+      if run_not_finished?(run) do
+        mark_run_as_lost(run)
+      end
+    end)
+  end
+
+  defp run_not_finished?(%{finished_at: nil, exit_reason: nil}), do: true
+
+  defp run_not_finished?(_), do: false
+
+  defp mark_run_as_lost(run) do
+    updated_run = Ecto.Changeset.change(run, exit_reason: "lost")
+    Repo.update!(updated_run)
```

Comment:
We can consider it's finished after this update like the attempt by updating the `finished_at`

Reply:
Actually maybe better not, right @stuartc? The `finished_at` could be just for the record of the update but `updated_at` would be enough and it keeps some sign/mark that it has never reported back to Lightning.

Reply:
Hmm, I don't really have a clear opinion about using `finished_at` when using `:lost`. Given what I said this morning about it, I would agree with you and that we don't know that it's not finished...

However in order to definitively tell how long it took to become lost, using `finished_at` is more reliable than `updated_at` since Ecto will set that at any point and we need to be more vigilant about making sure we don't alter the record.

So given that risk, I'd say we should use `finished_at` for now when marking as lost.

@benedictus-dev could you move the changeset code for `mark_run_as_lost` to the `Attempt` model. There you will find a bunch of changeset functions that do this kind of stuff and _additionally_ perform state transition validation.

Also please avoid bang functions (`update!`) without considering all the places these functions could be called - if there is no specific reason please let it return an error rather than throwing an exception.